libs/ts-rest/nest/src/lib/ts-rest-nest.ts

```diff
@@ -31,11 +31,9 @@ type AppRouterRequestShapes<T extends AppRouter> = Without<
 type AppRouterResponseShapes<T extends AppRouter> = Without<
   {
     [K in keyof T]: T[K] extends AppRoute
-      ? Promise<
```

Comment:
Having this wrapped in a promise means that you can't do:

```ts
const post: NestResponseShapes<typeof contract>['getPost'] = {...}
```

instead need to do:

```ts
const post: Awaited<NestResponseShapes<typeof contract>['getPost']> = {...}
```

Reply:
> Undoes change to NestResponseShapes which made return types a promise.

So what I understand from that is that `AppRouterResponseShapes` was not wrapped in a Promise and then was changed to be a Promise type. When was that "change" introduced?

If it was long ago, it's unfortunate, as the change back might break some users' code who are doing something like this
```typescript
async getPosts(
    @TsRestRequest()
    _: RequestShapes['getPosts']
  ): ResponseShapes['getPosts'] { ... }
```

Since the function would be expected a non-promise type but the async function is returning a promised type.

Reply:
Ah, I just checked and it was in the last PR. Let's ship it then!